Ported StereoVOExample_large and Edits to StereoVOExample

CodeXTL opened the pull request:
Ported the C++ StereoVOExample_large to a Python Notebook. The new notebook example is functionally the same as the C++ example with some additional explanation spread throughout. 

Update(s):
- The main difference is that data files are downloaded from the GTSAM Github rather than creating another data folder just for the Python examples. This also makes the notebook easier to use in Colab.
- Also made some minor changes to `StereoVOExample.ipynb`

Check(s):
- Checked the results of the C++ example and the newly ported notebook example, and they match
- Running `make check` passes all tests.

CodeXTL committed:
minor changes to StereoVOExample, update to StereoVOExample_large

CodeXTL committed:
small syntax changes in StereoVOExample, completed StereoVOExample_large

CodeXTL committed:
Small change in StereoVOExample_large

CodeXTL committed:
Cleaned up imports in StereoVOExample_large

dellaert commented on python/gtsam/examples/StereoVOExample.ipynb:
```diff
@@ -44,7 +44,19 @@
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "# Install necessary libraries\n",
```
- only if on colab

dellaert commented on python/gtsam/examples/StereoVOExample.ipynb:
```diff
@@ -44,7 +44,19 @@
+      "source": [
+        "# Install necessary libraries\n",
+        "%pip install -q gtsam-develop\n",
+        "%pip install -q numpy"
```
remove

dellaert commented on python/gtsam/examples/StereoVOExample.ipynb:
```diff
@@ -288,47 +283,10 @@
-          ]
-        }
-      ],
+      "outputs": [],
```
Make sure, before checking, to restart/run-all

dellaert commented on python/gtsam/examples/StereoVOExample_large.ipynb:
```diff
@@ -0,0 +1,286 @@
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
```
add `remove-cell`

dellaert commented on python/gtsam/examples/StereoVOExample_large.ipynb:
```diff
@@ -0,0 +1,286 @@
+      "outputs": [],
+      "source": [
+        "# Install necessary libraries\n",
+        "%pip install -q gtsam-develop\n",
```
if colab, only gtsam-develop

dellaert reviewed (commented):
Very cool, some comments, and some that we discussed in person

CodeXTL committed:
Fixed colab link, set some cells to hide, added Colab env check

CodeXTL committed:
set some cells to hide, added Colab env check, changed all data file load to use findExampleDataFile

dellaert commented:
Still some things that we discussed that are not there (separate parsing from adding to FG), and plotting, but I'll assume you're still working on this. Just request a re-review via Github when this is ready.

CodeXTL committed:
Large changes to StereoVOExample_large, with incomplete graphing

CodeXTL committed:
saved plotly draft in StereoVOExample_large

CodeXTL committed:
completed StereoVOExample_large notebook

CodeXTL committed:
Additional revisions to StereoVOExample notebook

CodeXTL committed:
Updated cell tags with minor stylistic changes

CodeXTL commented:
@dellaert Everything should be ready now!

Update(s):
- Added cell tags so unnecessary cells (copy right, Colab env check) are not rendered by MyST.
- Shifted Google Colab button to be right under the main title
- `StereoVOExample_large.ipynb` specific changes:
  - Made large revision to StereoVOExample_large to have more comprehensive explanation
  - Separated code for parsing and adding to FG
  - Added plotting to visually analyze the end result using Plotly
- `StereoVOExample.ipynb` specific changes
  - Minor updates/polishes to explanation

dellaert commented:
Very cool! Please make sure both notebooks are restarted and you have "run all"? I can't merge until this is done, as there will be no outputs on website.

CodeXTL commented:
@dellaert I did a quick check and ran all again, and all cells that outputs something should now have an output. As a side note, I also checked the notebooks by viewing them on Github, and I noticed that the plotly figures didn't render; I'm pretty sure this is an issue with Github because the figures render correctly on my local deployment of the MyST site and I can see the pyplot data in the output when I view the notebook as a text file. So, everything should be all good!
